apps/server/src/modules/tool/controller/dto/request/external-tool-config.params.ts

```diff
@@ -0,0 +1,12 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsString } from 'class-validator';
+
+export class ExternalToolConfigCreateParams {
+	@IsString()
+	@ApiProperty()
+	type!: string;
```

Comment:
try to make the field and class abstract and remove the validators, since you already (have to) use them in the concrete class.
```suggestion
export abstract class ExternalToolConfigCreateParams {
    abstract type: string;
}
```